geode-core/src/main/java/org/apache/geode/cache/query/internal/CompiledComparison.java

```diff
@@ -296,15 +296,23 @@ public boolean isRangeEvaluatable() {
   public int getSizeEstimate(ExecutionContext context) throws FunctionDomainException,
       TypeMismatchException, NameResolutionException, QueryInvocationTargetException {
     IndexInfo[] idxInfo = getIndexInfo(context);
-    if (idxInfo == null) {
-      // Asif: This implies it is an independent condition. So evaluate it first
-      // in filter operand
+
+    // Both operands are indexed, evaluate it first in the filter operand.
+    if (idxInfo != null && idxInfo.length > 1) {
       return 0;
     }
+
+    // Asif: This implies it is an independent condition. So evaluate it second in filter operand.
+    if (idxInfo == null) {
+      return 1;
+    }
+
     assert idxInfo.length == 1;
     Object key = idxInfo[0].evaluateIndexKey(context);
+
+    // Evaluate it second as we do with the independent condition if key was not found.
     if (key != null && key.equals(QueryService.UNDEFINED)) {
-      return 0;
+      return 1;
```

Comment:
I think if the key is UNDEFINED and we have indexes present, the index actually has a mapping for UNDEFINED so the lookup should itself be really fast so 0 might be ok

Reply:
Changed! 👍